test/langtools/tools/javac/processing/filer/TestOriginatingElements.java

```diff
@@ -0,0 +1,484 @@
+                                                                              JavaFileObject.Kind kind,
+                                                                              FileObject... originatingFiles) throws IOException {
+                    List.of(originatingFiles)
+                        .stream()
+                        .map(fo -> getInfo(fo))
+                        .forEach(testOutput::add);
+                    return super.getJavaFileForOutputForOriginatingFiles(location, className, kind, originatingFiles);
+                }
+                @Override
+                public FileObject getFileForOutputForOriginatingFiles(Location location,
+                                                                      String packageName,
+                                                                      String relativeName,
+                                                                      FileObject... originatingFiles) throws IOException {
+                    List.of(originatingFiles)
+                        .stream()
+                        .map(fo -> getInfo(fo))
+                        .forEach(testOutput::add);
+                    return super.getFileForOutputForOriginatingFiles(location, packageName, relativeName, originatingFiles);
+                }
+                private String getInfo(FileObject fo) {
+                    try {
+                        JavaFileObject jfo = (JavaFileObject) fo; //the test only expects JavaFileObjects here:
+                        JavaFileManager.Location location = jfo.getKind() == JavaFileObject.Kind.SOURCE
+                                ? StandardLocation.SOURCE_PATH
+                                : sjfm.getLocationForModule(StandardLocation.SYSTEM_MODULES, "java.base");
+                        String binaryName = inferBinaryName(location, jfo);
+                        return binaryName + "(" + jfo.getKind() + ")";
+                    } catch (IOException ex) {
+                        throw new AssertionError(ex);
+                    }
+                }
+            };
+            try {
+                String generatedData;
+                try (MemoryFileManager mfm = new MemoryFileManager(sjfm)) {
+                    compiler.getTask(null, mfm, null, null, null,
+                                     List.of(new ToolBox.JavaSource("package test; public class Generated2 {}")))
+                            .call();
+                    generatedData =
+                            Base64.getEncoder().encodeToString(mfm.getFileBytes(StandardLocation.CLASS_OUTPUT, "test.Generated2"));
+                }
+                List<String> options = List.of("-sourcepath", src.toString(),
+                                               "-processor", "TestOriginatingElements$P",
+                                               "-processorpath", System.getProperty("test.classes"),
```

Comment:
This may be enough. but note there is also `test.class.path` .... one is just the output directory, the other is the full path.